### DISCUSSION TITLE
iSpindel not pushing updates

### DISCUSSION THREAD

**Discussion by StodgyWaif** [mp-se/gravitymon#50] [Q&A]
Hello.
The data collection was going okay for about a day and half.
Missed some updates here and there but now there hasn't been any for about the last 7 hours.
I have the interval set to 900s.
Tried rebooting my Fermentrack server but still nothing.
Not positive but I think the last battery reading was between 4 and 4.1
This is on GravityMon v0.8
Any suggestions?

**Comment by mp-se** [comment]
The battery should last more than a month with those settings and that battery reading. The most common issue would be networking/wifi poor connection. In the last message that was sent to fermentrack what does the RSSI value say?  You can view that buy clicking on the questionmark in the upper corner and view the ispindle_raw_output.log, -55 to -70 is a good signal and below -80 it can have issues to connect. 

Some network routes will also keep a log of when the device was last connected, that could also be a place to check.

**Reply by StodgyWaif** [reply]
Thanks for the reply.
Here is the output:
```
('{"name" : "Yellow", "ID": "17479d", "token" : "fermentrack", "interval": '
 '900, "temperature": 36.0, "temp-units": "F", "gravity": 0.0000, "angle": '
 '56.71, "battery": 4.06, "rssi": -66, "corr-gravity": 0.0000, "gravity-unit": '
 '"G", "run-time": 1.0 }')
```
I checked my DHCP server and the last request by it was 8:42 this morning.
So I'm assuming that either it's not connecting to the Wifi or it's frozen.
I do have additional units so I was thinking of swapping it.
I can gather more info if you need anything.

**Reply by mp-se** [reply]
The signal strenght looks good. You can check the error log and post those entires, that might give some hints. 

If its frozen its due to some hardware problem so it does not wake up correctly. There is a watchdog that would reset it if there is a software problem.

You can check the voltage of the battery in case there is fault in the voltage calulation (wrong factor) that can show a higher voltage than the actual value.

**Reply by StodgyWaif** [reply]
Okay so it seems to be a hardware issue. I watched it for a while to see if the LED would flash and it never did so I restarted it.
The LED does not light up on the D1 Mini. (a clone version)
I checked the battery voltage and it's still at 4.06. I tried plugging the D1 into the PC with and without the switch turned on.
(I have removed the D1 diode)
No LED light at all and no response on the PC so I'm guessing the D1 is dead.
Any suggestions for reviving it? Or I just need to replace it?

Thanks again for the replies.

**Reply by mp-se** [reply]
I would check the soldering and also check if there is power to the D1 mini. Could be a bad soldering. 

If you have removed the diode you need to have the battery turned on for power or the device will not turn on. With the diode removed there is no power from that usb.  If you have it plugged in on the serial and turn on the switch the computer should make a sound that it has found a serial port.

**Reply by mp-se** [reply]
I might be dead if the flash memory has been corrupted but if thats the case you should be able to get it connected via serial at least.

**Reply by StodgyWaif** [reply]
Thanks for the suggestions.
I determined that the charging board has voltage at the B+ and B- pins but not the Out+ and Out- pins.
I'm assuming this isn't normal?

**Reply by mp-se** [reply]
B pins should come from the battery so there should be power. Out+ would the output to the ispindel (via the switch) so there should be voltage or the board would not start. Could be a faulty charge board you have there. 

https://www.ispindel.de/docs/circuit_diagram_en.html

**Comment by mp-se** [comment]
If you want to try out the 0.9 version I'm currently testing it. If you select the dev branch you can download the binaries from there. I will not add any more features only do testing and fix bugs.

**Reply by StodgyWaif** [reply]
Okay thanks.
I tried to use https://github.com/mp-se/gravitymon/tree/dev/bin/ as the OTA URL and reset the unit multiple times but it's not updating.
EDIT:
I don't know that much about GitHub but it seems like you can't access files on it this way.
I will try to do it locally instead.

**Reply by mp-se** [reply]
Github will try to redirect to another URL which the update feature does not understand so it will not work. I will see if I can find a workaround for this.

**Reply by mp-se** [reply]
I've managed to place a beta under the docs pages which works. When the real release is done you need to update that manually. Use this OTA url; https://mp-se.github.io/gravitymon/beta/

The changes can be viewed here; https://github.com/mp-se/gravitymon/blob/dev/src_docs/source/releases.rst

**Reply by StodgyWaif** [reply]
Thanks.
I ended up using a local web server to do the update to v0.9.
It's been running consistently on 3 units on my desktop for the last 24 hours.
I did notice that the Temp Adj setting now shows 32 instead of 0. (And the F Temp seems to be reading accurately at this setting)

**Reply by mp-se** [reply]
> Thanks. I ended up using a local web server to do the update to v0.9. It's been running consistently on 3 units on my desktop for the last 24 hours. I did notice that the Temp Adj setting now shows 32 instead of 0. (And the F Temp seems to be reading accurately at this setting)

You are right, its not working as it should. I will fix that! It's showig 0C converted to F which is 32. It should show the delta instead.

**Reply by mp-se** [reply]
I've fixed the bugs I have found so far. I'm starting to do some long term test on the latest build on the dev branch now. I nothing else is found this would be the next release in a week.
